### PR TITLE
haslogicaltreenode

### PR DESCRIPTION
**Pull request by derekpappas**
**Related issue**: 

**Type of change**: feature request

**Impact**:  API addition (no impact on existing code)

**Development Phase**: implementation

**Release Notes**

Adding a trait which will add the logical tree node from the parent to the child...

### CODE REVIEW

File: src/main/scala/diplomaticobjectmodel/HasLogicalTreeNode.scala [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package diplomaticobjectmodel
```
Comment (hcook) [inline src/main/scala/diplomaticobjectmodel/HasLogicalTreeNode.scala]: This should match all the other packages and be
`package freechips.rocketchip.diplomaticobjectmodel`

**Comment by hcook** [comment]
Why don't you you just go ahead and add this trait to everything that currently has a `val logicalTreeNode = ...` in it?

File: src/main/scala/diplomaticobjectmodel/HasLogicalTreeNode.scala (hunk 1 above)
Comment (derekpappas) [reply src/main/scala/diplomaticobjectmodel/HasLogicalTreeNode.scala]: fixed

**Comment by derekpappas** [comment]
We need to talk about replacing the pattern for adding nodes. Lets get this PR approved so that I can add the composable cache.

**Pull request merged**